src/services/AnalysisService.ts

```diff
@@ -0,0 +1,277 @@
+import StringUtilities from "../StringUtilities";
+import SOOSAnalysisApiClient, {
+  ICreateScanRequestContributingDeveloperAudit,
+  ICreateScanResponse,
+} from "../api/SOOSAnalysisApiClient";
+import { SOOS_CONSTANTS } from "../constants";
+import {
+  ContributingDeveloperSource,
+  ContributingDevelopersVariableNames,
+  IntegrationName,
+  OutputFormat,
+  ScanStatus,
+  ScanType,
+} from "../enums";
+import { soosLogger } from "../logging";
+import { sleep } from "../utilities";
+import * as FileSystem from "fs";
+import * as Path from "path";
+
+interface IRunOutputFormatParams {
+  clientId: string;
+  projectHash: string;
+  projectName: string;
+  branchHash: string;
+  scanType: ScanType;
+  analysisId: string;
+  outputFormat: OutputFormat;
+  sourceCodePath: string;
+  workingDirectory: string;
+}
+
+interface IStartScanParams {
+  clientId: string;
+  projectHash: string;
+  analysisId: string;
+  scanType: ScanType;
+  scanUrl: string;
+}
+
+interface IWaitForScanToFinishParams {
+  scanStatusUrl: string;
+  scanUrl: string;
+}
+
+interface ISetupScanParams {
+  clientId: string;
+  projectName: string;
+  branchName: string;
+  commitHash: string;
+  buildVersion: string;
+  buildUri: string;
+  branchUri: string;
+  integrationType: string;
+  operatingEnvironment: string;
+  integrationName?: IntegrationName;
+  appVersion: string;
+  scriptVersion: string;
+  contributingDeveloperAudit: ICreateScanRequestContributingDeveloperAudit[];
+  scanType: ScanType;
+}
+
+interface IUpdateScanStatusParams {
+  clientId: string;
+  projectHash: string;
+  branchHash: string;
+  scanType: ScanType;
+  analysisId: string;
+  status: ScanStatus;
+  message: string;
+}
+
+class AnalysisService {
+  private analysisApiClient: SOOSAnalysisApiClient;
+
+  constructor(analysisApiClient: SOOSAnalysisApiClient) {
+    this.analysisApiClient = analysisApiClient;
+  }
+
+  static create(apiKey: string, apiURL: string): AnalysisService {
```

Comment:
this create can go away, we are doing `const analysisService = new AnalysisService(soosAnalysisApiClient);` to not have two instances.
https://github.com/soos-io/soos-sca/pull/1/files#r1411169651